mathext/ell_carlson.go

```diff
@@ -0,0 +1,154 @@
+// This code is also available as a part of SLATEC Common Mathematical Library (http://netlib.org/slatec/index.html). Later, Carlson described
+// an improved version in http://dx.doi.org/10.1007/BF02198293 (also available at https://arxiv.org/abs/math/9409227).
+func EllipticRD(x, y, z float64) float64 {
+	const (
+		lower = 4.8095540743116787026618007863123676393525016818363e-103 // (5*2^-1022)^(1/3)
+		upper = 1 / lower
+		tol   = 9.0351169339315770474760122547068324993857488849382e-03 // (ε/5)^(1/8)
+	)
+	if x < 0 || y < 0 || math.IsNaN(x) || math.IsNaN(y) || math.IsNaN(z) {
+		return math.NaN()
+	}
+	if upper < x || upper < y || upper < z {
+		return math.NaN()
+	}
+	if x+y < lower || z < lower {
+		return math.NaN()
+	}
+
+	A0 := (x + y + 3*z) / 5
+	An := A0
+	Q := math.Max(math.Max(math.Abs(A0-x), math.Abs(A0-y)), math.Abs(A0-z)) / tol
+	xn, yn, zn := x, y, z
+	mul, s := 1.0, 0.0
+
+	for Q >= mul*math.Abs(An) {
+		xnsqrt, ynsqrt, znsqrt := math.Sqrt(xn), math.Sqrt(yn), math.Sqrt(zn)
+		lambda := xnsqrt*ynsqrt + ynsqrt*znsqrt + znsqrt*xnsqrt
+		s += 1 / (mul * znsqrt * (zn + lambda))
+		An = (An + lambda) * 0.25
+		xn = (xn + lambda) * 0.25
+		yn = (yn + lambda) * 0.25
+		zn = (zn + lambda) * 0.25
+		mul *= 4
+	}
+
+	X := (A0 - x) / (mul * An)
+	Y := (A0 - y) / (mul * An)
+	Z := -(X + Y) / 3
+	E2 := X*Y - 6*Z*Z
+	E3 := (3*X*Y - 8*Z*Z) * Z
+	E4 := 3 * (X*Y - Z*Z) * Z * Z
+	E5 := X * Y * Z * Z * Z
+
+	// http://dlmf.nist.gov/19.36.E2
+	return ((471240-540540*E2)*E5+(612612*E2-540540*E3-556920)*E4+E3*(306306*E3+E2*(675675*E2-706860)+680680)+E2*((417690-255255*E2)*E2-875160)+4084080)/(4084080*mul*An*math.Sqrt(An)) + 3*s
+}
+
+// EllipticF computes the Legendre's elliptic integral of the 1st kind F(\phi|m), 0≤m<1:
+//  F(\phi|m) = \int_{0}^{\phi} 1 / \sqrt{1-m\sin^2(\theta)} d\theta
+//
+// Legendre's elliptic integrals can be expressed as symmetric elliptic integrals, in this case:
+//  F(\phi|m) = \sin\phi R_F(\cos^2\phi,1-m\sin^2\phi,1)
+//
+// The definition of F(\phi,k) where k=\sqrt{m} can be found in NIST Digital Library of Mathematical
+// Functions (http://dlmf.nist.gov/19.2.E4).
+func EllipticF(phi, m float64) float64 {
+	s, c := math.Sincos(phi)
+	return s * EllipticRF(c*c, 1-m*s*s, 1)
+}
+
+// EllipticE computes the Legendre's elliptic integral of the 2nd kind E(\phi|m), 0≤m<1:
+//  E(\phi|m) = \int_{0}^{\phi} \sqrt{1-m\sin^2(\theta)} d\theta
+//
+// Legendre's elliptic integrals can be expressed as symmetric elliptic integrals, in this case:
+//  E(\phi|m) = \sin\phi R_F(\cos^2\phi,1-m\sin^2\phi,1)-(m/3)\sin^3\phi R_D(\cos^2\phi,1-m\sin^2\phi,1)
+//
+// The definition of E(\phi,k) where k=\sqrt{m} can be found in NIST Digital Library of Mathematical
+// Functions (http://dlmf.nist.gov/19.2.E5).
+func EllipticE(phi, m float64) float64 {
+	s, c := math.Sincos(phi)
+	x, y := c*c, 1-m*s*s
+	return s * (EllipticRF(x, y, 1) - (m/3)*s*s*EllipticRD(x, y, 1))
```

Comment:
Ah, thanks. 

For future reference, use backticks so the asterisks aren't turned into italized font.  `a*b`, not a*b